core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/catalog_json/CamelCatalogCacheManager.java

```diff
@@ -0,0 +1,241 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.fusesource.ide.camel.model.service.core.catalog_json;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.fusesource.ide.camel.model.service.core.catalog_json.components.Component;
+import org.fusesource.ide.camel.model.service.core.catalog_json.dataformats.DataFormat;
+import org.fusesource.ide.camel.model.service.core.catalog_json.eips.Eip;
+import org.fusesource.ide.camel.model.service.core.catalog_json.languages.Language;
+
+/**
+ * @author lhein
+ */
+public class CamelCatalogCacheManager {
+	
+	private static Map<CamelCatalogCoordinates, CamelCatalogCache> CACHE = new HashMap<>();
+	
+	/**
+	 * flushes the cache
+	 */
+	public static void flush() {
+		CACHE.clear();
+	}
+	
+	/**
+	 * returns the cached catalog for the given coordinates or a new catalog cache if not existing
+	 * 
+	 * @param coordinates	the coordinates
+	 * @return	the cached catalog or an empty one if not yet in cache
+	 */
+	public static CamelCatalogCache getCachedCatalog(CamelCatalogCoordinates coordinates) {
+		if (!CACHE.containsKey(coordinates)) {
+			CACHE.put(coordinates, new CamelCatalogCache());
+		}
+		return CACHE.get(coordinates);
+	}
+	
+	/**
+	 * removes the cached catalog for the given coordinates
+	 * 
+	 * @param coordinates	the coordinates
+	 */
+	public static void clearCachedCatalog(CamelCatalogCoordinates coordinates) {
+		if (CACHE.containsKey(coordinates)) {
+			CACHE.remove(coordinates);
+		}
+	}
+	
+	/**
+	 * @author lhein
+	 */
+	public static class CamelCatalogCoordinates {
```

Comment:
The classes are only used in the context of the cache manager. Therefore I found it best to keep them all together. But I am not opposed to changes if you got a better solution.